dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java

```diff
@@ -1230,23 +1230,19 @@ public Workflow manualRegister(@ApiParam(hidden = true) @Auth User user,
         @ApiParam(value = "Descriptor type", required = true) @QueryParam("descriptorType") String descriptorType,
         @ApiParam(value = "Default test parameter file path") @QueryParam("defaultTestParameterFilePath") String defaultTestParameterFilePath) {
 
-        // Validate descriptor path based on type
-        if ("nfl".equals(descriptorType) && !defaultWorkflowPath.endsWith("nextflow.config")) {
-            throw new CustomWebApplicationException(
-                    "Please ensure that the given workflow path '" + defaultWorkflowPath + "' is of type " + descriptorType
+        for (DescriptorLanguage typeItem : DescriptorLanguage.values()) {
+            if (typeItem.getShortName().equalsIgnoreCase(descriptorType)) {
+                // check that plugin is active
+                if (typeItem.isPluginLanguage() && !LanguageHandlerFactory.getPluginMap().containsKey(typeItem)) {
+                    throw new CustomWebApplicationException("plugin for " + typeItem.getShortName() + " is not installed",
+                        HttpStatus.SC_BAD_REQUEST);
+                }
+                if (typeItem.getDefaultPrimaryDescriptorExtensions().stream().noneMatch(defaultWorkflowPath::endsWith)) {
+                    throw new CustomWebApplicationException(
+                        "Please ensure that the given workflow path '" + defaultWorkflowPath + "' is of type " + descriptorType
                             + " and ends in the file nextflow.config", HttpStatus.SC_BAD_REQUEST);
```

Comment:
Not entirely grokking this whole flow, but should `nextflow.config` be hard-coded like this?